lib/pengine/rules.c

```diff
@@ -373,6 +373,10 @@ populate_hash(xmlNode * nvpair_list, GHashTable * hash, gboolean overwrite, xmlN
         if (pcmk__str_eq((const char *)an_attr->name, XML_CIB_TAG_NVPAIR, pcmk__str_none)) {
             xmlNode *ref_nvpair = expand_idref(an_attr, top);
 
+            if (ref_nvpair == NULL) {
```

Comment:
This affects more than logging, doesn't it? In particular, previously we wouldn't check `ref_nvpair` at all unless `name` or `value` was missing from `an_attr`.

Reply:
Good catch

Reply:
FYI `expand_idref()` will return `an_attr` if it has no `id-ref` attribute; we can only get NULL if there is an invalid `id-ref` (which can't happen with schema validation enabled). But it's a good idea to preserve existing behavior (for the case where schema validation is disabled and there's an invalid `id-ref` but a valid name and value).